waku/node/peer_manager/waku_peer_store.nim

```diff
@@ -32,126 +34,139 @@ type
-  ## Get the stored information of a given peer.
+proc getPeer*(wps: WakuPeerStore, peerId: PeerId): RemotePeerInfo =
   RemotePeerInfo(
     peerId: peerId,
-    addrs: peerStore[AddressBook][peerId],
+    addrs: wps.store[AddressBook][peerId],
     enr:
-      if peerStore[ENRBook][peerId] != default(enr.Record):
-        some(peerStore[ENRBook][peerId])
+      if wps[ENRBook][peerId] != default(enr.Record):
+        some(wps[ENRBook][peerId])
       else:
         none(enr.Record),
-    protocols: peerStore[ProtoBook][peerId],
-    agent: peerStore[AgentBook][peerId],
-    protoVersion: peerStore[ProtoVersionBook][peerId],
-    publicKey: peerStore[KeyBook][peerId],
-
-    # Extended custom fields
-    connectedness: peerStore[ConnectionBook][peerId],
-    disconnectTime: peerStore[DisconnectBook][peerId],
-    origin: peerStore[SourceBook][peerId],
-    direction: peerStore[DirectionBook][peerId],
-    lastFailedConn: peerStore[LastFailedConnBook][peerId],
-    numberFailedConn: peerStore[NumberFailedConnBook][peerId],
+    protocols: wps.store[ProtoBook][peerId],
+    agent: wps.store[AgentBook][peerId],
+    protoVersion: wps.store[ProtoVersionBook][peerId],
+    publicKey: wps.store[KeyBook][peerId],
+    connectedness: wps.store[ConnectionBook][peerId],
+    disconnectTime: wps.store[DisconnectBook][peerId],
+    origin: wps.store[SourceBook][peerId],
+    direction: wps.store[DirectionBook][peerId],
+    lastFailedConn: wps.store[LastFailedConnBook][peerId],
+    numberFailedConn: wps.store[NumberFailedConnBook][peerId],
   )
 
-proc getWakuProtos*(peerStore: PeerStore): seq[string] =
-  ## Get the waku protocols of all the stored peers.
-  let wakuProtocols = toSeq(peerStore[ProtoBook].book.values())
-    .flatten()
-    .deduplicate()
-    .filterIt(it.startsWith("/vac/waku"))
-  return wakuProtocols
+proc addPeer*(wps: WakuPeerStore, peer: RemotePeerInfo) =
+  wps.store[AddressBook][peer.peerId] = peer.addrs
+  wps.store[ProtoBook][peer.peerId] = peer.protocols
+  wps.store[AgentBook][peer.peerId] = peer.agent
+  wps.store[ProtoVersionBook][peer.peerId] = peer.protoVersion
+  wps.store[KeyBook][peer.peerId] = peer.publicKey
+  wps.store[ConnectionBook][peer.peerId] = peer.connectedness
+  wps.store[DisconnectBook][peer.peerId] = peer.disconnectTime
+  wps.store[SourceBook][peer.peerId] = peer.origin
+  wps.store[DirectionBook][peer.peerId] = peer.direction
+  wps.store[LastFailedConnBook][peer.peerId] = peer.lastFailedConn
+  wps.store[NumberFailedConnBook][peer.peerId] = peer.numberFailedConn
+  if peer.enr.isSome():
+    wps.store[ENRBook][peer.peerId] = peer.enr.get()
+
+proc delete*(wps: WakuPeerStore, peerId: PeerId) =
+  # Delete all the information of a given peer.
+  wps.store.del(peerId)
 
 # TODO: Rename peers() to getPeersByProtocol()
-proc peers*(peerStore: PeerStore): seq[RemotePeerInfo] =
-  ## Get all the stored information of every peer.
+proc peers*(wps: WakuPeerStore): seq[RemotePeerInfo] =
```

Comment:
I wonder if we should also remane those to `getPeers` or `getWakuPeers` to make them more explicit and easier to lookup